### PR TITLE
made RELEASE mode default, added debug optimization flag

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

Changed CMakeLists.txt to default to release and add debug optimization flag for debug mode. 

## Related Issue

https://github.com/USGS-Astrogeology/ISIS3/issues/3387

## How Has This Been Tested?

Ran make in verbose mode to confirm `-O3` was being used with `-DCMAKE_BUILD_TYPE=RELEASE` and with node build type var. Same with `-DCMAKE_BUILD_TYPE=DEBUG` and confirmed `-OG` was being used.  

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/CMakeLists.txt [hunk 1]
```diff
@@ -186,6 +186,11 @@ endif()
  string(REPLACE ";" " " FLAGS_STR "${thirdPartyCppFlags}")
  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} ${FLAGS_STR}" )
 
+ if(NOT CMAKE_BUILD_TYPE)
+     set(CMAKE_BUILD_TYPE "RELEASE")
+ endif()
+
+ set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -OG")
```
Comment (Kelvinrr) [inline isis/CMakeLists.txt:193]: CMake already adds the `-O3` flag when mode=release so only debug mode needed an update.

**Review by jessemapel — APPROVED** [review]

**Comment by jessemapel** [comment]
Is this going into 3.8.1?

**Comment by Kelvinrr** [comment]
Presumably yes

**Comment by jessemapel** [comment]
I'm going to patch this onto 3.8 then

**Pull request merged**